### PR TITLE
Conan made some annoying changes in v1.9 that require more updates

### PR DESCRIPTION
**Pull request by DavidZemon**
FYI: Conan v1.9 switched from a registry.txt to a registry.json file and tries to automatically upgrade. As part of that upgrade, it will try to _remove_ your old .txt file. If the .txt file is mounted in a Docker container, Conan will completely fail to run. So... make sure you upgrade your _native_ Conan instance (`sudo -H pip3 install --upgrade conan`) and then run Conan (no args necessary, just `conan` will do) to perform the upgrade on your host before invoking any Docker/Conan builds.

### CODE REVIEW

**Review by lancemjennings — APPROVED** [review]

**Comment by DavidZemon** [comment]
TeamCity + Conan is broken until this is merged. @pwalkr, I'd like at least your eyes on this before merging. Everyone else is mostly just an "FYI, Conan is changing"

File: start.sh [hunk 1]
```diff
@@ -3,11 +3,13 @@
...
-if [ ! -s "${HOME}/.conan/registry.txt" ] ; then
-    cat "${HOME}/.conan/registry.template.txt" >> "${HOME}/.conan/registry.txt"
+if [[ ! -s "${HOME}/.conan/registry.json" ]] ; then
+    conan remote add ci "https://artifactory.redlion.net/artifactory/api/conan/conan-local"
+    conan remote add conan-center "https://conan.bintray.com"
+    conan remote add bintray-community "https://api.bintray.com/conan/conan-community/conan"
 fi
 
-if [ "${uid}" -a "${gid}" ] ; then
+if [[ "${uid}" && "${gid}" ]] ; then
```
Comment (pwalkr) [inline start.sh:6]: Below uses `((1000 != ${uid} || 1000 != ${gid} ))` instead of `-o`. Granted there's other differences, but why this change?

File: start.sh [hunk 2]
```diff
@@ -3,11 +3,13 @@
 # Note: set user using `--env uid=XXXX --env gid=XXXX`, instead of using
 # docker's `--user` flag
 
-if [ ! -s "${HOME}/.conan/registry.txt" ] ; then
-    cat "${HOME}/.conan/registry.template.txt" >> "${HOME}/.conan/registry.txt"
+if [[ ! -s "${HOME}/.conan/registry.json" ]] ; then
+    conan remote add ci "https://artifactory.redlion.net/artifactory/api/conan/conan-local"
```
Comment (pwalkr) [inline start.sh]: This makes the whole toolchain much less generic. If we have to `-v`-mount credentials, can we mount in remotes as well?

File: start.sh (hunk 1 above)
Comment (DavidZemon) [reply start.sh:6]: I don't know bash syntax very well, but the bash plugin i use in my IDEs suggested this quickfix, and I'm all for anything that gives me the more familiar `&&` syntax over `-a`

File: start.sh (hunk 2 above)
Comment (DavidZemon) [reply start.sh]: Mounting the registry file is encouraged, as shown in the `docker-native` script. But, if someone _doesn't_ mount in the registry file, I want it to still work.

**Review by pwalkr — APPROVED** [review]
